SignalServiceKit/src/Storage/TSStorageManager.m

```diff
@@ -43,7 +51,29 @@ + (instancetype)sharedManager {
     return sharedManager;
 }
 
-- (void)setupDatabaseWithSafeBlockingMigrations:(void (^_Nonnull)(void))safeBlockingMigrationsBlock
+- (instancetype)initStorage
+{
+    self = [super initStorage];
+
+    if (self) {
+        _dbReadConnection = self.newDatabaseConnection;
+        _dbReadWriteConnection = self.newDatabaseConnection;
+
+        OWSSingletonAssert();
+    }
+
+    return self;
+}
+
+- (void)resetStorage
+{
+    _dbReadConnection = nil;
```

Comment:
It's an edge case, but is it safe to set these to nil? They aren't marked as nullable nor lazily initialized.

Reply:
We've always done it this way.  I consider this a work in progress towards closing (some) databases in the background.